R/gh_token.R

```diff
@@ -83,7 +83,21 @@
 gh_token <- function(api_url = NULL) {
   api_url <- api_url %||% default_api_url()
   token_env_var <- paste0("GITHUB_PAT_", slugify_url(api_url))
-  get_first_token_found(c(token_env_var, "GITHUB_PAT", "GITHUB_TOKEN"))
+  candidates <- c(
+    token_env_var,
+    if (is_github_dot_com(api_url)) c("GITHUB_PAT", "GITHUB_TOKEN")
+  )
+  val <- get_first_token_found(candidates)
+  if (val != "" || !can_load("credentials")) {
+    return(val)
+  }
+
+  set_github_pat2(api_url)
```

Comment:
The main new function, which is adapted from `credentials::set_github_pat()`, hence it's temporary name.